ext.php

```diff
@@ -15,38 +15,80 @@
 
 class ext extends \phpbb\extension\base
 {
+	/**
+	 * @var array An array of installation error messages
+	 */
+	protected $errors = [];
+
 	/**
 	 * Check whether or not the extension can be enabled.
 	 *
-	 * @return bool
+	 * @return bool|array True if can be enabled. False if not, or an array of error messages in phpBB 3.3.
 	 */
 	public function is_enableable()
 	{
-		return $this->php_requirements() && $this->phpbb_compatible();
+		// Check requirements
+		$this->phpbb_requirement();
+		$this->php_requirement();
+		$this->ziparchive_exists();
+
+		return count($this->errors) ? $this->enable_failed() : true;
 	}
 
 	/**
-	 * Check PHP requirements
+	 * Check phpBB 3.2.0 minimum requirement.
 	 *
-	 * Requires PHP 5.4.0 or greater
-	 * Requires PHP ZipArchive binary
+	 * @return void
+	 */
+	protected function phpbb_requirement()
+	{
+		if (phpbb_version_compare(PHPBB_VERSION, '3.2.0', '<'))
+		{
+			$this->errors[] = 'PHPBB_VERSION_ERROR';
+		}
+	}
+
+	/**
+	 * Check PHP 5.4.0 minimum requirement.
 	 *
-	 * @return bool
+	 * @return void
 	 */
-	protected function php_requirements()
+	protected function php_requirement()
 	{
-		return phpbb_version_compare(PHP_VERSION, '5.4.0', '>=') && class_exists('ZipArchive');
+		if (phpbb_version_compare(PHP_VERSION, '5.4.0', '<'))
```

Comment:
It won't be installed on phpBB < 3.2, but 3.2 can be only installed with PHP 5.4.7+.
So this check is probably can be dropped here as it is always satisfied.

Reply:
PHP 5.4.7 is due to the s9e library which updated at some point, previous 3.2 versions run on 5.4.0.

Reply:
Also, it doen't hurt just to check :)